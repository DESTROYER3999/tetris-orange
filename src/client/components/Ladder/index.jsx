import { connect } from "react-redux";
import LadderSub from "./subcomponent";

const mapStateToProps = state => ({
  highscores: state.game.highscores
});

const Ladder = connect(
  mapStateToProps,
  null
)(LadderSub);

export default Ladder;
