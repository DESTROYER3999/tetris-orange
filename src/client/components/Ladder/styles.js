import styled, { css } from "styled-components";

export const StyledMainContainer = styled.div`
  flex: 1;
  padding: 20px;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px 60px 2px 2px;
  display: flex;
  flex-direction: column;
  // justify-content: center;
  align-items: center;
  h2 {
    flex: 1;
  }
`;

export const StyledList = styled.div`
  flex: 3;
  width: 100%;
  display: flex;
  flex-direction: column;
`;

export const StyledListItem = styled.div`
  flex: 1;
  display: flex;
  ${props =>
    props.index % 2 === 0 &&
    css`
      background-color: rgba(0, 0, 0, 0.2);
    `};
`;

export const StyledListItemContent = styled.div`
  display: flex;
  justify-content: center;
  div {
    font-size: 1.5vh;
    align-self: center;
  }
  flex: 1;
`;
