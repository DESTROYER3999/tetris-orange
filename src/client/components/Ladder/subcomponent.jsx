import React from "react";
import PropTypes from "prop-types";
import {
  StyledMainContainer,
  StyledList,
  StyledListItem,
  StyledListItemContent
} from "./styles";

const LadderSub = ({ highscores }) => {
  return (
    <StyledMainContainer id="ladder">
      <h2>High Scores</h2>
      <StyledList id="highscore-list">
        {highscores.map((highscore, index) => {
          return (
            <StyledListItem id="list-item" key={index} index={index}>
              <StyledListItemContent id="content">
                <div>{index + 1 + "."}</div>
              </StyledListItemContent>
              <StyledListItemContent id="content">
                <div>{highscore.score}</div>
              </StyledListItemContent>
              <StyledListItemContent id="content">
                <div>{highscore.playerName}</div>
              </StyledListItemContent>
            </StyledListItem>
          );
        })}
      </StyledList>
    </StyledMainContainer>
  );
};

LadderSub.propTypes = {
  highscores: PropTypes.array.isRequired
};

export default LadderSub;
