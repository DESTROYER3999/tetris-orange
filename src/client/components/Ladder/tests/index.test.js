import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import Ladder from "../index";
import socket from "../../../services/socket-api";

const mockStore = configureStore();
const initialState = {
  game: {
    highscores: []
  }
};

const store = mockStore(initialState);

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<Ladder />", () => {
  test("Ladder rendering with store without crashing", () => {
    const wrapper = shallow(<Ladder store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
