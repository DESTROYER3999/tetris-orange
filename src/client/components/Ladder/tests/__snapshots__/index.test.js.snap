// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ladder /> Ladder rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <LadderSub
    dispatch={[Function]}
    highscores={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
